File: PBOZAKI/php/Produk.php
<?php

class Produk
{
    private string $namaProduk;
    private float $harga;
    private int $stok;

    public function __construct(
        string $namaProduk,
        float $harga,
        int $stok
    ) {
        if (trim($namaProduk) === '') {
            throw new InvalidArgumentException(
                "Nama produk tidak boleh kosong."
            );
        }

        if ($harga < 0) {
            throw new InvalidArgumentException(
                "Harga tidak boleh negatif."
            );
        }

        if ($stok < 0) {
            throw new InvalidArgumentException(
                "Stok tidak boleh negatif."
            );
        }

        $this->namaProduk = $namaProduk;
        $this->harga = $harga;
        $this->stok = $stok;
    }

    public function tambahStok(int $jumlah): void
    {
        if ($jumlah <= 0) {
            throw new InvalidArgumentException(
                "Jumlah stok harus lebih dari 0."
            );
        }

        $this->stok += $jumlah;
    }

    public function kurangiStok(int $jumlah): void
    {
        if ($jumlah <= 0) {
            throw new InvalidArgumentException(
                "Jumlah stok yang dikurangi harus lebih dari 0."
            );
        }

        if ($jumlah > $this->stok) {
            throw new InvalidArgumentException(
                "Stok tidak mencukupi."
            );
        }

        $this->stok -= $jumlah;
    }

    public function tampilkanInfo(): void
    {
        echo "=== DATA PRODUK ===" . PHP_EOL;
        echo "Nama Produk : " . $this->namaProduk . PHP_EOL;
        echo "Harga       : Rp" . $this->harga . PHP_EOL;
        echo "Stok        : " . $this->stok . PHP_EOL;
    }
}

File: PBOZAKI/program/Produk.java
public class Produk {
    private String namaProduk;
    private double harga;
    private int stok;

    public Produk(String namaProduk, double harga, int stok) {
        if (namaProduk == null || namaProduk.isBlank()) {
            throw new IllegalArgumentException(
                "Nama produk tidak boleh kosong."
            );
        }

        if (harga < 0) {
            throw new IllegalArgumentException(
                "Harga tidak boleh negatif."
            );
        }

        if (stok < 0) {
            throw new IllegalArgumentException(
                "Stok tidak boleh negatif."
            );
        }

        this.namaProduk = namaProduk;
        this.harga = harga;
        this.stok = stok;
    }

    public void tambahStok(int jumlah) {
        if (jumlah <= 0) {
            throw new IllegalArgumentException(
                "Jumlah stok yang ditambahkan harus lebih dari 0."
            );
        }

        stok += jumlah;
    }

    public void kurangiStok(int jumlah) {
        if (jumlah <= 0) {
            throw new IllegalArgumentException(
                "Jumlah stok yang dikurangi harus lebih dari 0."
            );
        }

        if (jumlah > stok) {
            throw new IllegalArgumentException(
                "Stok tidak mencukupi."
            );
        }

        stok -= jumlah;
    }

    public void tampilkanInfo() {
        System.out.println("=== DATA PRODUK ===");
        System.out.println("Nama Produk : " + namaProduk);
        System.out.println("Harga       : Rp" + harga);
        System.out.println("Stok        : " + stok);
    }
}